// postmybuild/src/scanner.js
import fs from "fs";
import path from "path";

const MAX_README_LENGTH = 500;

export function scanProject(rootDir = process.cwd()) {
  const result = {
    name: null,
    description: null,
    techStack: [],
    hasReadme: false,
    readmeSnippet: null,
  };

  // ---- package.json ----
  const packageJsonPath = path.join(rootDir, "package.json");

  if (fs.existsSync(packageJsonPath)) {
    try {
      const pkg = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));

      result.name = pkg.name || null;
      result.description = pkg.description || null;

      const deps = {
        ...pkg.dependencies,
        ...pkg.devDependencies,
      };

      result.techStack = Object.keys(deps || {});
    } catch (err) {
      console.error("⚠️ Could not read package.json");
    }
  }

  // ---- README ----
  const files = fs.readdirSync(rootDir);
  const readmeFile = files.find(f => f.toLowerCase() === "readme.md");

  if (readmeFile) {
    try {
      const readmeContent = fs.readFileSync(path.join(rootDir, readmeFile), "utf-8");
      result.hasReadme = true;
      result.readmeSnippet = readmeContent
        .replace(/\s+/g, " ")
        .slice(0, MAX_README_LENGTH);
    } catch (err) {
      console.error("⚠️ Could not read README file");
    }
  }


  return result;
}
